File: php/tests/JewelleryStorageTest.php
<?php

namespace Tests\NecklaceRefactoringKata;

use NecklaceRefactoringKata\Earring;
use NecklaceRefactoringKata\EarringType;
use NecklaceRefactoringKata\Jewel;
use NecklaceRefactoringKata\JewelleryStorage;
use NecklaceRefactoringKata\Packer;
use PHPUnit\Framework\TestCase;

class JewelleryStorageTest extends TestCase
{
    function test_pack_earring_stud(): void
    {
        $storage = new JewelleryStorage();
        $item = new Earring(Jewel::Amber, EarringType::Stud);

        Packer::pack($item, $storage);

        $this->assertContains($item, $storage->box->topShelf);
    }
}

File: kotlin/src/test/kotlin/org/sammancoaching/necklace/JewelleryStorageTest.kt
package org.sammancoaching.necklace

import org.junit.jupiter.api.Test

internal class JewelleryStorageTest {
  @Test
  internal fun `pack earring stud`() {
    val storage = JewelleryStorage()
    val earring = Earring(EarringType.Stud, Jewel.Amber)
    pack(earring, storage)
    TODO("check it packed it correctly")
  }
}
